File: Codeforces/1032A.py
from sys import stdin
from collections import Counter

data=stdin.read().split()
n,k=map(int,data[:2])
x=Counter(data[2:])  

d=max(x.values())    ## Maximum number of one type of utensils remaining
s=((d+k-1)//k)*k    ## Total number of dishes served is ceil(d/k)*k

count=0
for v in x.values():
    count+=s-v
print(count)

File: Codeforces/1032A.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    //Input Variables
    int n,k;
    cin>>n>>k;

    vector<int> a(n);

    //Helper variable
    vector<int> x(101,0);        // Stores occurenece of each (viables as n<=100)
    
    for (int i=0;i<n;i++) {
        cin>>a[i];
        x[a[i]]++;
    }

    sort(x.begin(),x.end());
    int d=x[100];               // maximum number of vessel available of one type
    // cout<<"d:"<<d<<"\n";        // Debug

    int s=(d+k-1)/k*k;            // Ceil of d/k *k is max number of dishes serves
    // cout<<"s:"<<s<<"\n";        // Debug

    int count=0;                //Output Variable
    for (int  i :x) {
        if (i==0) continue;
        count+=(s-i);
    }
    cout<<count;
    return 0;
}
